C/include/c4Replicator.h

```diff
@@ -40,12 +40,15 @@ extern "C" {
 
     /** The possible states of a replicator. */
     typedef C4_ENUM(int32_t, C4ReplicatorActivityLevel) {
+        /* EXTERNAL STATES */
         kC4Stopped,     ///< Finished, or got a fatal error.
-        kC4Stopping,    ///< Stopping or going offline (INTERNAL STATE)
         kC4Offline,     ///< Connection failed, but waiting to retry. */
         kC4Connecting,  ///< Connection is in progress.
         kC4Idle,        ///< Continuous replicator has caught up and is waiting for changes.
-        kC4Busy         ///< Connected and actively working.
+        kC4Busy,         ///< Connected and actively working.
+        
+        /* INTERNAL STATES */
+        kC4Stopping,    ///< Stopping or going offline
```

Comment:
In term of the Lite-Core API, I think putting kC4Stopping is better. Is the provide a guideline to CBL to change instead of moving kC4Stopping to the end of the enum? 

When moving the kC4Stopping like that, it doesn't allow to do something like `status > kC4Stopping` to check whether the replicator is in stopping or stopped state.

Reply:
I think that, perhaps, this is what Jim means by a "code smell".  Assuming that we know the order and exact number of states seems fragile and easy to break.  I think that a good way forward is to assure that, once a state is assigned a number, that number never changes.  Changing the number is changing the API and it is my opinion that adding or removing internal states is not a good reason for doing that.

Reply:
I looked over and didn't find any situations where we need to do that.  Since we aren't mapping C4 to CBL for this state I thought it would be better in the long run to split into external and internal sections.

Reply:
👍🏻